lib/containers/k8s.pm

```diff
@@ -32,11 +34,18 @@ sub install_k3s {
     assert_script_run("export INSTALL_K3S_BIN_DIR=" . get_var('K3S_BIN_DIR')) if (get_var('K3S_BIN_DIR'));
     assert_script_run("export INSTALL_K3S_CHANNEL=" . get_var('K3S_CHANNEL')) if (get_var('K3S_CHANNEL'));
     # github.com/k3s-io/k3s#5946 - The kubectl delete namespace helm-ns-413 command freezes and does nothing
-    assert_script_run("curl -sfL https://get.k3s.io | sh -s - --disable=metrics-server");
+    # https://github.com/k3s-io/k3s/issues/6052
     # Note: The install script starts a k3s-server by default, unless INSTALL_K3S_SKIP_START is set to true
-    sleep(20);    # Wait one iteration interval before checking because the server needs some time to boot-up
+    assert_script_run("curl -sfL https://get.k3s.io | INSTALL_K3S_SKIP_START=true INSTALL_K3S_VERSION=v1.24.4+k3s1 sh -s - --disable=metrics-server");
```

Comment:
Hardcoded versions are asking for trouble. Is there a particular reason why we need to do this? If https://github.com/k3s-io/k3s/issues/6052 was the reason, then the issue suggests that the problem has been resolved in the meantime.